src/commands/services/add.cc

```diff
@@ -51,7 +51,20 @@ void add_service( Zypper & zypper, const ServiceInfo & service )
         // bsc#1203715: Support (re)adding a service with the same URL.
         MIL << "Service '" << service.alias() << "' exists with same URL '" << service.url() << "'" << endl;
         zypper.out().info( str::Format(_("Service '%1%' with URL '%2%' already exists. Just updating the settings.")) % service.alias() % service.url() );
-        manager.modifyService( service.alias(), service );
+        // bsc#1252744: Update service enabled/refresh settings but keep all
+        // the remembered repo data. They will be updated on the next refresh.
+        bool mustModify = false;
+        if ( exists.enabled() != service.enabled() ) {
+          exists.setEnabled( service.enabled() );
+          mustModify = true;
+        }
+        if ( exists.autorefresh() != service.autorefresh() ) {
+          exists.setAutorefresh( service.autorefresh() );
+          mustModify = true;
+        }
+        if ( mustModify ) {
+          manager.modifyService( exists.alias(), exists );
+        }
```

Comment:
Isn't that the wrong way around? exists is the service already present in the RepoManager, while service is the new one... shouldn't we set autorefresh and enabled in the new service and then call modifyService with it?

Reply:
It's IMO right.

- In this case `exists` and `service` have same alias and same URL. 
- `service` has the desired `enabled` and `autorefresh` values specified on the CLI but no repo states.
- `exists`  has the existing repo states.

We formerly set the `service` state, losing the repo states (modifyService -> service).
We now adjust the desired `enabled` and `autorefresh` values and keep the repo states (modifyService -> exists).
We could vice versa transfer all repo states into `service` and (modifyService -> service).